### PR TITLE
=act #17425 Document NonBlockingBoundedMailbox to be single-consumer

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #17425

File: akka-actor/src/main/scala/akka/dispatch/Mailbox.scala [hunk 1]
```diff
@@ -386,7 +386,10 @@ class NodeMessageQueue extends AbstractNodeQueue[Envelope] with MessageQueue wit
   }
 }
 
-//Discards overflowing messages into DeadLetters
+/**
+ * Lock-free bounded non-blocking multiple-producer single-consumer queue.
+ * Discards overflowing messages into DeadLetters.
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: should we mention this(discarding) in the doc?

**Comment by drewhk** [comment]
LGTM

File: akka-actor/src/main/scala/akka/dispatch/Mailbox.scala (hunk 1 above)
Comment (patriknw) [inline akka-actor/src/main/scala/akka/dispatch/Mailbox.scala]: done, thx

**Pull request merged**